src/workflows/bidsFFX.m

```diff
@@ -73,14 +73,16 @@ function bidsFFX(action, opt, funcFWHM)
 
           saveAndRunWorkflow(matlabbatch, batchName, opt, subID);
 
-          plot_power_spectra_of_GLM_residuals( ...
-                                              getFFXdir(subID, funcFWHM, opt), ...
-                                              opt.metadata.RepetitionTime);
+          if opt.glmQA.do
+            plot_power_spectra_of_GLM_residuals( ...
+                                                getFFXdir(subID, funcFWHM, opt), ...
+                                                opt.metadata.RepetitionTime);
 
-          deleteResidualImages(getFFXdir(subID, funcFWHM, opt));
+            deleteResidualImages(getFFXdir(subID, funcFWHM, opt));
```

Comment:
not sure, but this means that if I do not run the `glmQA`, then the residuals will not be merged in a 4D image and then not deleted?

Reply:
Not quite.

I had added this other bit below: the residuals will not be written if you don't run the GLM QA because this is the only thing they are used for so far.

```
  if ~opt.glmQA.do
    writeResiduals = false();
  end
  matlabbatch{end}.spm.stats.fmri_est.write_residuals = writeResiduals;
```

Reply:
ok so that is the part I was not understanding, thanks!